### PR TITLE
Add Ldap autoconfiguration support

### LINKED ISSUE #2645: Add LdapAutoConfiguration
LDAP Support is missing... I think that it's time to add support for it.

### PR DESCRIPTION
**Pull request by eddumelendez**
Fix #2645

### CODE REVIEW

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ldap/LdapProperties.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+public class LdapProperties {
+
+	/**
+	 * LDAP urls
+	 */
+	@Size(min = 1)
+	private String[] urls = new String[0];
+
+	/**
+	 * Base suffix from which all operations should origin.
```
Comment (johnktims) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ldap/LdapProperties.java]: originate*

File: spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/ldap/LdapAutoConfigurationTests.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import org.springframework.ldap.core.LdapTemplate;
+
+import static org.hamcrest.Matchers.hasEntry;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for {@link LdapAutoConfiguration}.
+ *
+ * @author Marcel Overdijk
```
Comment (johnktims) [inline spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/ldap/LdapAutoConfigurationTests.java]: Typo?

**Comment by mdiskin** [comment]
Thanks for this PR I have added it locally until sprint boot provides it.
Any chance it could support pooling as part of the auto configuration too?

The other item that typically goes with LDAP configuration is a FilterBasedLdapUserSearch bean that could be autoconfigured.

**Comment by eddumelendez** [comment]
Closing this PR in favor of new third-party project [ldap-spring-boot](https://github.com/eddumelendez/ldap-spring-boot)

**Pull request closed without merging**